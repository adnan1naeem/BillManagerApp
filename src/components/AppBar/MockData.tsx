
export const Data = {

  backbuttonTrue: true,

  backbuttonfalse: false,
  title: "BillManager",
  style: { color: 'yellow' }

}