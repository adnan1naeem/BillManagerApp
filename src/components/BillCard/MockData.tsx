
export const Data = {
  month: "Jan",
  budget: 2000,
  name: "Bills"
}
