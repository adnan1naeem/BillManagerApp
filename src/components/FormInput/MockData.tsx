export const Data = {

  IconColor: "black",
  IconName: "md-checkmark-circle",
  IconSize: 20,
  label: "ADNANA",
  onValueChange: () => 'string'
}



