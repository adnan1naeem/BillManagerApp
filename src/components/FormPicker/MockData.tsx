export const Data = {

  IconColor: "black",
  IconName: "md-checkmark-circle",
  IconSize: 20,
  label: "label",
  onValueChange: () => 'string',
  Items: [{ label: 'AC', value: 'AC' }, { label: 'bulb', value: 'bulb' }]
}



