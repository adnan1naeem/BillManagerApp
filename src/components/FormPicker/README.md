#  FormPicker Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native    |
| :----------------------: | :----------------------: | :----------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_check_mark: |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                             |
| :----: | :---------: | :-----------------------------------------: |
| TBD    | TBD         | ![FormPicker](./Screenshots/FormPicker.png) |
|        |

## Props

| Name          | Type     | Default | Description                               |
| :------------ | :------- | :------ | :---------------------------------------- |
| label         | string   |         | It is used for adding label  in component |
| onValueChange | function |         | It is used for changing input value       |
| value         | string   |         | It is used for adding input values        |
| IconSize      | number   |         | It is used for adding Size of Icon        |
| IconName      | string   |         | It is used for adding Name of icon        |
| IconColor     | string   |         | It is used for adding Color of icon       |


## How to use

```react
import React from 'react';
<FormPicker
      label={'Name'}
      onValueChange={(e)=>e}
      IconColor={"blue"}
      IconSize={20}
      IconName={"arrow-left"}
      Items={[{label:'Name',value:'Name'}]}
      value={e.value}
    />
```

## Tests
