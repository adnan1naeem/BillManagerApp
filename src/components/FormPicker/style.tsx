export const styles = {

  container: {
    flex:1,
    display: 'flex',
    flexDirection: 'column',
    paddingTop:20
  },
  Name: {
    marginLeft: 20,
    marginTop: 5
  },
  IconContainer: {

    display: 'flex',
    flexDirection: 'row'
  },
  Iconstyles:{
    alignSelf: 'center' 
  },
  Picker:{
     flex: 1 
 },
 
}