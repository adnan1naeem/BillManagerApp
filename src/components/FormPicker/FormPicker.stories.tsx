import React from 'react';
import { storiesOf } from '@storybook/react-native';
import FormPicker from './FormPicker.component';
import { withKnobs, text, } from '@storybook/addon-knobs/react';
import { View } from 'react-native';
import { Data } from "./MockData"

storiesOf('FormPicker', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs)

  .add('FormPicker story', () => (
    <FormPicker
      IconSize={Data.IconSize}
      IconName={Data.IconName}
      IconColor={Data.IconColor}
      label={Data.label}
      handleChange={Data.onValueChange}
      Items={Data.Items}
    />
  ))



