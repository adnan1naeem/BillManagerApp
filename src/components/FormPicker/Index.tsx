import FormPicker from  "./FormPicker.component";
export default FormPicker; 