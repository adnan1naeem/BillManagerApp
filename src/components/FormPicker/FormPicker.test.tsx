import * as React from 'react'
import { shallow, configure } from 'enzyme';
import FormPicker from "./FormPicker.component";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Data } from "./MockData"
module.exports = {};
configure({ adapter: new Adapter() })

describe('FormPicker Input unit Tests', () => {

  test('if FormPickerInput renders correctly ', () => {
    shallow(<FormPicker
      label={Data.label}
      Items={Data.Items}
      handleChange={Data.onValueChange}
      IconName="ios-arrow-dropdown-circle"
      IconSize={0}
      IconColor="green"

    />);
  });

  test('FormPickerInput SnapShot Test', () => {
    const tree = renderer.create(<FormPicker
      label={Data.label}
      Items={Data.Items}
      handleChange={Data.onValueChange}
      IconName="ios-arrow-dropdown-circle"
      IconSize={0}
      IconColor="green"

    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
