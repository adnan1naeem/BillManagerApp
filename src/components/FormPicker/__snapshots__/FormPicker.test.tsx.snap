// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPicker Input unit Tests FormPickerInput SnapShot Test 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 20,
        "marginTop": 5,
      }
    }
  >
    label
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <View
      style={
        Object {
          "borderBottomColor": "#4381FF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "AC",
              "textColor": undefined,
              "value": "AC",
            },
            Object {
              "label": "bulb",
              "textColor": undefined,
              "value": "bulb",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
